src/components/ExportToCognitoS3Dialog/index.js

```diff
@@ -0,0 +1,213 @@
+import React, { useEffect, useState } from "react"
+import SimpleDialog from "../SimpleDialog"
+import DataTable from "react-data-table-component"
+import useActiveDatasetManager from "../../hooks/use-active-dataset-manager"
+import isEmpty from "lodash/isEmpty"
+import datasetManagerCognito from "udt-dataset-managers/dist/CognitoDatasetManager"
+import useAuth from "../../utils/auth-handlers/use-auth"
+import { TextField } from "@material-ui/core"
+const redText = { color: "orange" }
+const tableStyle = {
+  marginLeft: "auto",
+  marginRight: "auto",
+  width: "100%",
+}
+
+const expandedAnnotationsColumns = [
+  { name: "Annotations", selector: "annotation" },
+  { name: "Last Modified", selector: "lastModified", sortable: true },
+]
+
+const columns = [{ name: "Projects", selector: "folder", sortable: true }]
+
+const customStyles = {
+  headCells: {
+    style: {
+      paddingLeft: "10px",
+      "font-weight": "bold",
+    },
+  },
+  cells: {
+    style: {
+      paddingLeft: "25px",
+    },
+  },
+}
+
+const ExpandedRow = ({ data }) => {
+  const { rowAnnotations } = data
+  return (
+    <>
+      <DataTable
+        style={{
+          boxSizing: "border-box",
+          paddingLeft: "50px",
+          paddingRight: "10px",
+        }}
+        dense
+        striped
+        noHeader
+        columns={expandedAnnotationsColumns}
+        data={rowAnnotations}
+        noDataComponent='Make sure the project has "samples" folder'
+        pagination={rowAnnotations.length > 10}
+        paginationPerPage={10}
+        paginationRowsPerPageOptions={[10, 20, 25, 50, 100, 200]}
+        customStyles={customStyles}
+      />
+    </>
+  )
+}
+
+export default ({ open, onClose }) => {
+  const [dm, setDm] = useState()
+  const { authConfig } = useAuth()
+  const [projects, setProjects] = useState()
+  const [activeDatasetManager] = useActiveDatasetManager()
+  const [nameProjectToCreate, setNameProjectToCreate] = useState("")
+  const [nameProjectExist, setNameProjectExist] = useState(false)
+  const getCurrentDataset = async () => {
+    var ds = await activeDatasetManager.getDataset()
```

Comment:
try to avoid loading entire datasets into memory because when theyre big these function calls can take several minutes (even for datasets with only a few thousand samples). Try using useDatasetProperty

Reply:
If I get the samples it the same problem ?

Reply:
I think I don't have a choice for the export to load the whole dataset because i export all of it

Reply:
I will close it since i don't have answer.